samples/star-wars-api/HttpHandlers.fs

```diff
@@ -8,277 +8,350 @@ open System.Text.Json.Serialization
+                    req.EnableBuffering ()
+                    let body = req.Body
+                    let buffer = Array.zeroCreate 1
+                    // All we need to ask for is the first byte.
+                    let! bytesRead = body.ReadAsync (buffer, 0, 1)
+                    body.Seek (0, SeekOrigin.Begin) |> ignore
+                    return bytesRead > 0
+                }
+
+        /// Check for the conditions that would make this an introspection query
+        let isRequestEmpty (req : HttpRequest) (logger : ILogger)=
+            if req.Method = HttpMethods.Get then
+                logger.LogTrace ("Request is GET. Must be an introspection query")
+                true |> Task.FromResult
             else
-                return! executeOperation()
-        } |> ofTaskIResult ctx
+                task {
+                    let! hasBody = hasRequestBody req
+                    if not hasBody then
+                        logger.LogTrace ("Request is not GET but has no body. Must be an introspection query")
+                        return true
+                    else
+                        logger.LogTrace ("Request is not GET and has a body")
+                        return false
+                }
+
+        task {
+            let! requestIsEmpty = isRequestEmpty request logger
+
+            /// Returns true if the given HttpRequest is multi-part.
+            let isMultipartRequest (req : HttpRequest) =
+                not (System.String.IsNullOrEmpty(req.ContentType)) && req.ContentType.Contains("multipart/form-data")
+
+            if requestIsEmpty then
+                let! result = executeIntrospectionQuery ValueNone
+                let response = result |> toResponse
+                return Results.Ok response
+    
+            elif isMultipartRequest request then
+
+                /// Look for a header specifying the boundary string used in a multi-part form submission and return the string if it exists.
+                let getMultipartRequestBoundary (req : HttpRequest) =
+                    req.Headers.GetCommaSeparatedValues("Content-Type")
+                    |> Seq.map (fun v -> v.TrimStart())
+                    |> Seq.tryFind (fun v -> v.Contains("boundary"))
+                    |> Option.map (fun v -> v.Remove(0, v.IndexOf('=') + 1))
+                    |> Option.map (fun v -> v.Trim('"'))
+
+                match getMultipartRequestBoundary request with
+                | Some boundary ->
+
+                    let copyBodyToMemory (req : HttpRequest) =
+                        let ms = new MemoryStream(4096)
+                        req.Body.CopyTo(ms)
+                        ms.Position <- 0L
+                        ms
+
+                    return! task {
+                        use ms = copyBodyToMemory request
+                        let reader = MultipartReader(boundary, ms)
+                        let! multiRequest = reader |> MultipartRequest.read ctx.RequestAborted
```

Comment:
I don't know what was the idea of copy to `MemoryStream` but we have a buffered stream on line 252 so se can just seek back if we have to. Hence simply pass the body